java/arcs/android/devtools/DevWebServerImpl.kt

```diff
@@ -49,6 +53,10 @@ object DevWebServerImpl : DevWebServer, NanoWSD("localhost", 33317) {
         super.closeAllConnections()
     }
 
+    fun addOnOpenWebsocketCallback(callback: () -> String) {
+        onOpenSocketCallbacks.add(callback)
+    }
```

Comment:
Does this need to be public?

Reply:
Made it internal so it could still be called from DevToolsService